### PR TITLE
update spec for router-mode

### PR DESCRIPTION
**Pull request by ajssmith**

### CODE REVIEW

**Review by grs — APPROVED** [review]
LGTM!

File: client/site_config_inspect.go [hunk 1]
```diff
@@ -76,7 +85,16 @@ func (cli *VanClient) SiteConfigInspect(ctx context.Context, input *corev1.Confi
...
-		result.Spec.Ingress = types.IngressLoadBalancerString
+		// check for deprecated key
+		if clusterLocal, ok := siteConfig.Data["cluster-local"]; ok {
+			if clusterLocal == "true" {
+				result.Spec.Ingress = types.IngressNoneString
+			} else {
+				result.Spec.Ingress = types.IngressLoadBalancerString
+			}
+		} else {
+			result.Spec.Ingress = types.IngressLoadBalancerString
```
Comment (fgiorgetti) [inline client/site_config_inspect.go:96]: @ajssmith should we assume loadbalancer here or use `GetIngressDefault()`?

File: cmd/skupper/skupper.go [hunk 2]
```diff
@@ -210,11 +209,17 @@ installation that can then be connected to other skupper installations`,
 			}
 
 			if routerModeFlag.Changed {
-				options := []string{routerModeInteriorValue, routerModeEdgeValue}
+				options := []string{string(types.TransportModeInterior), string(types.TransportModeEdge)}
```
Comment (fgiorgetti) [inline cmd/skupper/skupper.go:212]: TransportModeEdge seems to be missing the TransportMode type at `types.go`

File: client/router_create.go [hunk 3]
```diff
@@ -399,7 +400,7 @@ func (cli *VanClient) GetRouterSpecFromOpts(options types.SiteConfigSpec, siteId
 		Name:          "http",
 		ContainerPort: types.TransportLivenessPort,
 	})
-	if !options.IsEdge {
+	if isEdge {
```
[the remark this answers is not in the document]
Comment (ajssmith) [reply client/router_create.go:403]: Great catch!

File: client/site_config_inspect.go (hunk 1 above)
Comment (ajssmith) [reply client/site_config_inspect.go:96]: Very good suggestion. I was keeping the previous default but now that we have GetIngressDefault() we can make a better selection.

File: cmd/skupper/skupper.go (hunk 2 above)
Comment (ajssmith) [reply cmd/skupper/skupper.go:212]: hmm, I see that it is present

**Review by fgiorgetti — APPROVED** [review]
It is working fine now Andy.
I have observed a few issues with http and mongodb tests, but they are
not related with this change (as I am already seeing it happening with
older skupper images).

**Pull request merged**